metrics/xrootd_metrics.go

```diff
@@ -1740,7 +1792,90 @@ func HandleSummaryPacket(packet []byte) error {
 	return nil
 }
 
-// handleOSSPacket processes the OSS plugin stats
+func updateIntCounter(new int, old int, counter prometheus.Counter) int {
+	incBy := math.Max(0, float64(new-old))
+	counter.Add(incBy)
+	if new < old {
+		return old
+	}
+	return new
+}
+func updateFloat64Counter(new float64, old float64, counter prometheus.Counter) float64 {
+	incBy := new - old
+	if incBy < 0 {
+		incBy = 0
+	}
+	counter.Add(incBy)
+	if new < old {
+		return old
+	}
+	return new
+}
```

Comment:
If you haven't played around with go's "generics", this smells like exactly the type of thing they're meant for. You should take a look and see whether this could be written as one function on both ints and floats 🤓